e2e/benchmarks/services_test.go

```diff
@@ -0,0 +1,276 @@
+// +build e2e
+
+// Runs benchmarks using the Null provider and working with Consul setup with
+// multiple Services or Service Instances.
+
+package benchmarks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/command"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const configFile = "config.hcl"
+const tempDirPrefix = "tmp_"
+
+// Benchmark N Services Instance (all the same service)
+func Benchmark1Instance(b *testing.B) {
+	benchmarkInstances(b, 1)
+}
+func Benchmark100Instances(b *testing.B) {
+	benchmarkInstances(b, 100)
+}
+func Benchmark1000Instances(b *testing.B) {
+	benchmarkInstances(b, 1000)
+}
+func benchmarkInstances(b *testing.B, N int) {
+	srv := testutils.NewTestConsulServerHTTPS(b, "../../testutils")
+	defer srv.Stop()
+	path, cleanup := setupServiceInstances(b, srv, N)
```

Comment:
N service instances are registered with Consul as the name "service", however CTS for the `benchmarkInstances` are monitoring the service named "stress" which does not end up querying all of those instances. Can you update either the name they are registered with Consul as or update the service name the CTS task is monitoring?

Reply:
Saw this in the last one. Kind of hoped you missed it so I could fix it as this is getting embarrassing.

Reply:
Going to fix this so it uses the same variable. Eliminate the worry of keeping them in sync.

Reply:
Hey... I thought HTTP/2 would be useful here but then it ended up not being. Should have thought about that more at the time as it would have indicated this issue as it does hit the issue with 1000 services. So I switched the MultiServices ones to use HTTP/2 while the Instances ones will stick with HTTP/1.

Reply:
New changes look good! Thanks for fixing it.

Are you observing different Consul API behavior between the two MultiServices and Instances over HTTP/1? It's not clear to me how the Consul HTTP limit doesn't affect 1000 instances but does for 1000 services when they're both using the same endpoint.

Reply:
Wouldn't the Instances case be 1 watched service with 1000 entries and so only need the 1 connection?

Reply:
Oh I thought this was on the topic of registering the services within `setupServiceInstances` and not the client for watching.

Reply:
You are right, I switched context there sort of combining this and the changes related to the previous comment. Made sense to me at the time but was a bit of a jump. Sorry for the confusion.